tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +725,31 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
+                        # If heartbeat extension was not negotiated, then
+                        # we send unexpected_message fatal alert
+                        if not self.heartbeat:
+                            for result in self._sendError( \
+                                    AlertDescription.unexpected_message,
+                                            "Heartbeat was not negotiated"):
+                                yield result
+
+                        try:
+                            heartbeat_message = Heartbeat().parse(p)
```

Comment:
If I remember correctly, the RFC says that malformed heartbeat messages are to be ignored, isn't that the case?